packages/shared/lib/sdk/sync.ts

```diff
@@ -941,16 +944,31 @@ export class NangoSync extends NangoAction {
                 continue;
             }
 
+            hasErrors.push({ data: record, validation });
             metrics.increment(metrics.Types.RUNNER_INVALID_SYNCS_RECORDS);
 
+            if (this.runnerFlags?.validateSyncRecords) {
+                break;
+            }
```

Comment:
what is this `if` for?

Reply:
otherwise it just logs everything*

Reply:
hasErrors is never gonna have more than 1 element then if we always early break

Reply:
yes sorry it's maybe not clear, this flag is only set to true in CLI

Reply:
ah ok. so in CLI it breaks out after the first validation error